### ISSUE TITLE
Wrong process variable value order with type 'List'

### ISSUE THREAD

**Issue by azinchenko-sc** [OpusCapita/grailsflow-core-plugin#26]
From time to time we have wrong order in '**List**' variable. It is not always can be reproduced(every 2-7 workflow with 6+ approvers).

Everything is OK in database storage:
![image](https://user-images.githubusercontent.com/25082370/36150996-a7499d8c-10d6-11e8-8737-6ab5f45dcb02.png)

So there is a problem with data retrieval during variable initialization.

```
case LIST:
    if (variable) {
        Set<ProcessVarListItem> items = variable?.items
        List values = []
        items?.each() {
            values << getConvertedValue(it.content, variable?.subTypeName, variable)
        }
        return values
    }
```

The problem is that **PersistentSet** will not always have correct order so we need to save sorted result into **List**.

**Comment by azinchenko-sc** [comment]
e4e3b59  was reverted.
Adjusted to retrieve data from db in 'PersistentSortedSet' collection instead of 'PersistentSet' and further sorting.

**Comment by ivouchak-sc** [comment]
The reason: wrong comparator, order can't be taken into account for a new created items.

**Comment by ivouchak-sc** [comment]
The issue was resolved in 1.7.15
